### ISSUE TITLE
docker save does not work with large (6GB+) image

### ISSUE THREAD

**Issue by janpieterk** [docker/for-mac#2308]
### Expected behavior

$ docker save large_image > large_image.tar
should result in a tar file with the image.

### Actual behavior

command never finishes, seems to abort after 30 seconds, no spinning wheel in Terminal tab anymore, no prompt. Stays this way for days (left it running over the weekend). Return to prompt only possible with Ctrl-C. Result is empty large_image.tar file. Nothing ever gets written to it.

### Information

Diagnostic ID: BB1908E0-B859-454A-A2D4-26D7C333F799
Info from Console app below. See [ERROR] at 12-12-17 08:38:50,971.

12-12-17 08:37:36,350 Docker[528]: Pidfile com.docker.hyperkit references Pid 789 but expected path /Applications/Docker.app/Contents/Resources/bin/hyperkit <> actual path '': assuming pid re-use
12-12-17 08:37:39,378 Docker[533]: com.docker.hyperkit: [INFO] Resized file to 131072 clusters (16777216 sectors)
12-12-17 08:37:39,378 Docker[533]: com.docker.hyperkit: [INFO] block recycler starting with keep_erased = 2048
12-12-17 08:37:39,391 Docker[533]: com.docker.hyperkit: [INFO] image has 0 free sectors and 131069 used sectors
12-12-17 08:37:44,380 Docker[533]: com.docker.hyperkit: [INFO] block recycler: triggering background flush: 131069 used; 0 junk; 0 erased; 0 available; 0 copies; 0 roots; 0 Copying; 0 Copied; 0 Flushed; 0 Referenced; max_cluster = 131071
12-12-17 08:38:20,970 Docker[533]: com.docker.hyperkit: [INFO] Allocator: 131069 used; 0 junk; 0 erased; 0 available; 0 copies; 0 roots; 0 Copying; 0 Copied; 0 Flushed; 0 Referenced; max_cluster = 131071
12-12-17 08:38:20,970 Docker[533]: com.docker.hyperkit: [INFO] Allocator: file contains cluster 0 .. 131071 will enlarge file to 0 .. 131583
12-12-17 08:38:20,970 Docker[533]: com.docker.hyperkit: [INFO] resize: adding available clusters (Node ((x 131072) (y 131583) (l Empty) (r Empty) (h 1) (cardinal 512)))
12-12-17 08:38:50,971 Docker[533]: com.docker.hyperkit: [ERROR] write sector = 4528184 length = 49152: I/O deadline exceeded
12-12-17 08:38:50,971 Docker[533]: com.docker.hyperkit: [INFO] (((description "Anonymous client 1875")

**Comment by Jazzepi** [comment]
How much free memory do you have before running this command (and while running it)? I wonder if docker is doing something dumb like not-streaming the commit, and trying to load it all into memory before committing it to disk.

**Comment by janpieterk** [comment]
Free memory before and during running: about 450 MB (output of vm_stat). Free memory doesn't significantly change while running the command. Also, Docker memory usage looks like this:

Docker	9,1 MB
com.docker.osx.hyperkit.linux	2,8 MB	
com.docker.driver.amd64-linux	7,1 MB
hyperkit	149,6 MB	
com.docker.db	31,5 MB	
vpnkit	14,1 MB
com.docker.osxfs	3,7 MB

Both before and while running the command (output copied from Activity Monitor).

**Comment by janpieterk** [comment]
Version 17.12.0-ce-mac46 (21698) solved this problem, image saved successfully.

**Comment by XtineInWa** [comment]
I am running Windows Server 2016, with docker version 18.03.0-ce. I've tried unsuccessfully to save an image that is a copy of a sql express container. So far, I've left it alone for an hour and it still hasn't returned to a prompt (finished saving to tar file). I found the temporary files under ProgramData/Docker/tmp and see that their sizes has increased super slowly, but it has finally reached the 13.2 GB size and is not increasing.... This image very base image is SQL Express. 

Any ideas on why it is taking over an hour and still hasn't saved to a completed tar file? Thanks!
